client/amqp/amqp.go

```
@@ -191,8 +197,13 @@ func (a *AMQP) ReceiveTaskResponse(ctx context.Context) (orbital.TaskResponse, e
 		return orbital.TaskResponse{}, err
 	}
 
-	resp, err := a.codec.DecodeTaskResponse(msg.GetData())
+	data := msg.GetData()
+	resp, err := a.codec.DecodeTaskResponse(data)
 	if err != nil {
+		slog.Error("failed to decode TaskResponse", "data", string(data), "error", err)
+		if errAck := a.receiver.AcceptMessage(ctx, msg); errAck != nil {
```

I think we can dry that. Meaning we are just logging the error here without returning and let line 210 take care of the ack. We would of course need to save the error. Meaning when decode fails but ack succeeds we still need to return an error